### PR TITLE
feat: Support loading manifest from artifacts for resource templates. Fixes #5878

### LINKED ISSUE #5878: Creating A Resource Using Git Artifact
## Summary 

I have a workflow with 2 steps. The first is simply doing a git clone in order to create a git artifact. The repository I clone has a workflow yaml in it. I then want to use the workflow yaml in the next step, where I have a resource template. That is, the resource template should create the workflow which was in the git repo cloned in step 1! 

## Diagnostics

What Kubernetes provider are you using? GKE

What version of Argo Workflows are you running? 3.0.1

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: possible-bug-
spec:
  entrypoint: all-steps
  templates:
  - name: all-steps
    steps:
    - - name: step-1-create-git-artifact
        template: git-clone 
    - - name: step-2-create-workflow
        template: create-resource
        arguments:
          parameters:
          - name: resource-manifest
            value: "steps.step-1-create-git-artifact.outputs.parameters.manifest"
  - name: git-clone
    inputs:
      artifacts:
      - name: argo-source
        path: /src
        git:
          repo: https://github.com/argoproj/argo-workflows.git
          revision: master
    container:
      image: golang:1.10
      workingDir: /src
      command: [sh, -c]
      args: ["sleep 1;
              cat examples/hello-world.yaml > /tmp/hello-world.yaml"]
    outputs:
      parameters:
        - name: manifest
          valueFrom:
            default: "hello-world.yaml didnt work"
            path: "/tmp/hello-world.yaml"
  - name: create-resource
    inputs:
      parameters:
      - name: resource-manifest
    resource:
      action: create
      manifest: "{{inputs.parameters.resource-manifest}}"
```

```
Paste the logs from the workflow controller:
 Error (exit code 1): error: error validating "/tmp/manifest.yaml": error validating data: invalid object to validate; if you choose to ignore these errors, turn validation off with --validate=false  
```

I have also seen this error when changing the yamls:

```
create-resource master-resource-creator-lhvgb-729962375  2s    Error (exit code 1): error: error parsing /tmp/manifest.yaml: error converting YAML to JSON: yaml: mapping values are not allowed in this context
```

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by iven**
Signed-off-by: Iven Hsu <ivenvd@gmail.com>
Fixes #5878

Don't bother creating a PR until you've done this:

* [x] Run `make pre-commit -B` to fix codegen, lint, and commit message problems.

Create your PR as a draft.

* Your PR needs to pass the required checks before it can be approved. If the check is not required (e.g. E2E tests) it
  does not need to pass.
* Once required tests have passed, you can make it "Ready for review".
* Say how how you tested your changes. If you changed the UI, attach screenshots.

Tips:

* If changes were requested, and you've made them, then dismiss the review to get it looked at again.
* Add you organization to [USERS.md](https://github.com/argoproj/argo-workflows/blob/master/USERS.md) if you like.
* You can ask for help!

### CODE REVIEW

File: workflow/executor/executor.go [hunk 1]
```diff
@@ -263,6 +263,9 @@ func (we *WorkflowExecutor) StageFiles() error {
 		filePath = common.ExecutorScriptSourcePath
 		body = []byte(we.Template.Script.Source)
 	case wfv1.TemplateTypeResource:
+		if we.Template.Resource.ManifestPath != "" {
+			return nil
```
Comment (sarabala1979) [inline workflow/executor/executor.go:267]: `ManifestPath`. should be.optional.  it should not return. here

**Review by sarabala1979 — CHANGES_REQUESTED** [review]

File: workflow/executor/executor.go (hunk 1 above)
Comment (iven) [reply workflow/executor/executor.go:267]: I think it's optional. If `ManifestPath` not provided, it will not return.

**Comment by sarabala1979** [comment]
Added @alexec for spec changes

**Review by alexec — COMMENTED** [review]
what about `manifestFrom`? the we could have:

```yaml
manifestFrom: 
  path: ...
  configMap: ..
  etc
```

**Comment by wsj0000** [comment]
> what about `manifestFrom`? the we could have:
> 
> ```yaml
> manifestFrom: 
>   path: ...
>   configMap: ..
>   etc
> ```

That also what I need. When the resource manifest is large, the action will blocked on the size limits of parameter (which is the 
size limits of `annotations`). No container generated and the `artifact` is also not avaliable.

**Comment by iven** [comment]
I'm not working on this any more, @book987 will continue the work.

**Comment by book987** [comment]
> @book987 Any update on this?

sorry no update yet, I'll work on it this weekend.

**Comment by terrytangyuan** [comment]
This is still needed. Also note that we will only support loading a single resource from manifests.

**Comment by book987** [comment]
> This is still needed. Also note that we should only support loading a single resource from manifests.

Hi sorry for late, I've create a new pr for this #7980

**Pull request closed without merging**